src/main/java/io/reactivex/internal/operators/single/SingleDelay.java

```diff
@@ -39,33 +39,57 @@ protected void subscribeActual(final SingleObserver<? super T> s) {
 
         final SequentialDisposable sd = new SequentialDisposable();
         s.onSubscribe(sd);
-        source.subscribe(new SingleObserver<T>() {
-            @Override
-            public void onSubscribe(Disposable d) {
-                sd.replace(d);
+        source.subscribe(new OnSubscribeObserver(sd, s));
+    }
+
+    private class OnSubscribeObserver implements SingleObserver<T> {
```

Comment:
These could be named more aptly, for example, `DelayObserver` so it shows up as `io.reactivex.interna.operators.single.SingleDelay$DelayObserver`.